k2/csrc/fsa_algo.cc

```diff
@@ -147,4 +149,48 @@ void Connect(const Fsa &a, Fsa *b, std::vector<int32_t> *arc_map /*=nullptr*/) {
   }
 }
 
+void ArcSort(const Fsa &a, Fsa *b,
+             std::vector<int32_t> *arc_map /*= nullptr*/) {
+  CHECK_NOTNULL(b);
+  b->arc_indexes = a.arc_indexes;
+  b->arcs.clear();
+  b->arcs.reserve(a.arcs.size());
+  if (arc_map != nullptr) arc_map->clear();
+
+  using ArcWithIndex = std::pair<Arc, int32_t>;
+  std::vector<int32_t> indexes(a.arcs.size());  // index mapping
+  int32_t n = 0;
+  std::generate(indexes.begin(), indexes.end(), [&n]() { return n++; });
```

Comment:
`std::iota` https://en.cppreference.com/w/cpp/algorithm/iota

No lambda function and no extra function invoke for `std::itoa`.

`int32_t n = 0; ` can be removed when using `std::itoa`.

-----

`++n` is preferred over `n++`.

Reply:
Sure

Reply:
`int32_t n = -1` is required if using `++n` here, which is not so readable.

Anyway, changed to iota